src/routes/api/tlv/+server.ts

```diff
@@ -27,12 +27,24 @@ export const GET = async ({ fetch }) => {
     return new Response('[]', { headers: { 'Content-Type': 'application/json' } });
   }
 
-  const dripsTokenHoldingsRes = await (
-    await fetch(
-      `https://api.etherscan.io/api?module=account&action=addresstokenbalance&address=0xd0Dd053392db676D57317CD4fe96Fc2cCf42D0b4&page=1&offset=100&apikey=${etherscanApiKey}`,
-    )
-  ).json();
-  const dripsTokenHoldings = etherscanTokensResponseSchema.parse(dripsTokenHoldingsRes.result);
+  const driptsTokenHoldingRes = await fetch(
+    `https://api.etherscan.io/api?module=account&action=addresstokenbalance&address=0xd0Dd053392db676D57317CD4fe96Fc2cCf42D0b4&page=1&offset=100&apikey=${etherscanApiKey}`,
+  );
+  if (!driptsTokenHoldingRes.ok) {
+    const errorContent = await driptsTokenHoldingRes.text();
+    // eslint-disable-next-line no-console
+    console.error('Etherscan returned error response', errorContent);
+    return new Response('[]', { headers: { 'Content-Type': 'application/json' } });
```

Comment:
Why return an empty array? it usually returns a number as a string. on the actual `default-explore-page.svelte` it checks for `typeof tlv === 'number'`. I think ideally:

- this endpoint returns either nothing or an error 500 with empty body in case there's an etherscan error
- the logic calling in the TLV value catches an error response from this endpoint, and returns `null` in that case
- `default-explore-page.svelte` explicitly accepts either `number` or `null` as `tlv` prop, and conditionally renders the TLV `if tlv !== null`

wdyt?

Reply:
Yes down to change! I was following the logic in the previous lines that return an empty array if there's no etherscan key. It looks like it eventually ends up in a Intl formatting method which returns $0.0 when given an empty array...weird! 

I think returning null is great in these cases. If the tlv endpoint returns a 500, the whole loading blows up as it is currently.

Reply:
I see, didn't catch that! Yeah, if there's no etherscanApiKey i think it also should be returning null and handle it the same way as when there's an error — produce a server-side log, and gracefully hide the TLV card.

Reply:
Yeah that makes total sense, will do